frontend/src/components/organizations/OrganizationInfo.vue

```diff
@@ -216,55 +221,95 @@ export default {
         return {}
       },
     },
-    loading: {
+    uploadedDocuments: {
+      type: Array,
+      default: () => {
+        return []
+      },
+    },
+    loadingAll: {
+      type: Boolean,
+      default: false,
+    },
+    loadingInclusionPolicy: {
       type: Boolean,
       default: false,
     },
   },
+  emits: ['saveInclusionPolicyData'],
   data() {
     return {
-      isFormComplete: false,
-      organizationReadOnly: undefined,
-      organizationEdit: undefined,
-      localLoading: false,
-      rules,
+      organizationEdit: {},
       editMode: false,
+      uploadedDocumentsEdit: [],
+      documentsToUpload: [],
+      documentsToDelete: [],
+      showUploadDocumentsAlert: false,
     }
   },
-  computed: {
-    ...mapState(useAuthStore, ['userInfo']),
-  },
   async updated() {
-    this.organizationReadOnly = JSON.parse(JSON.stringify(this.organization))
+    this.initializeData()
   },
   methods: {
-    async update() {
-      this.$refs.organizationForm?.validate()
-      if (!this.isFormComplete) {
+    async initializeData() {
+      this.organizationEdit = { ...this.organization, hasInclusionPolicy: this.organization.hasInclusionPolicy || false }
+      this.uploadedDocumentsEdit = JSON.parse(JSON.stringify(this.uploadedDocuments))
+    },
+
+    documentsExistOrBeingAdded() {
+      return !isEmpty(this.documentsToUpload) || !isEmpty(this.uploadedDocumentsEdit)
+    },
+
+    async save() {
+      if (this.organizationEdit?.hasInclusionPolicy && !this.documentsExistOrBeingAdded()) {
+        this.showUploadDocumentsAlert = true
         return
       }
-      try {
-        this.localLoading = true
-        await OrganizationService.updateOrganization(this.userInfo?.organizationId, this.organizationEdit)
-        this.organizationReadOnly = { ...this.organizationEdit }
-        this.editMode = false
-        this.setSuccessAlert('Organization information updated successfully')
-      } catch (error) {
-        this.setFailureAlert('Failed to get your organization information', error)
-      } finally {
-        this.localLoading = false
+      if (!this.organizationEdit.hasInclusionPolicy && this.organization.hasInclusionPolicy) {
+        this.documentsToDelete = this.uploadedDocumentsEdit.map((document) => document.documentId)
       }
+      this.$emit('saveInclusionPolicyData', this.organizationEdit, this.documentsToUpload, this.documentsToDelete)
     },
 
     toggleEditMode() {
-      this.organizationEdit = JSON.parse(JSON.stringify(this.organization))
       this.editMode = !this.editMode
+      this.showUploadDocumentsAlert = false
+      if (!this.editMode) {
+        this.organizationEdit.hasInclusionPolicy = this.organization.hasInclusionPolicy
+      }
```

Comment:
Is this required? Is this to ensure that the hasInclusionPolicy flag is kept in sync?

Reply:
Replaced with initalizeData() as discussed.